end2end/tests/reportBuilder.spec.ts

```diff
@@ -2,30 +2,208 @@ import { test, expect } from '@playwright/test';
+    await expect(page.getByLabel('Sort by Current Status')).toBeVisible();
+    await expect(page.getByLabel('Sort by Type')).toBeVisible();
+});
+
+test("User Redirect to SearchFilter Page on Modify Filter", async ({ page }) => {
+    await page.waitForTimeout(5000);
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHJfNDA0UyhFGhg5dxwGMCRgNRBhNBhIpABGW0LyLJywAHkAMwq4fTsVIA%3D%3D%3D&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgXRiEHeln1Gi6stU8AukA");
+    await page.getByRole('button', { name: 'Modify Search' }).click();
+    await page.getByRole('cell', { name: 'Resolved' }).locator('a').click();
+    await page.getByRole('cell', { name: 'Current Status' }).locator('a').click();
+    await page.getByRole('option', { name: 'Data Field' }).click();
+    await page.getByRole('cell', { name: 'CONTAINS' }).locator('a').click();
+    await page.getByRole('group', { name: 'Advanced Search Options' }).click();
+    await page.locator('a').filter({ hasText: 'Any standard data field' }).click();
+    await page.getByRole('option', { name: 'LEAF Developer Console: Supervisor' }).click();
+    await page.getByRole('button', { name: 'Next Step' }).click();
+    await page.getByText('Approval History').click();
+    await expect(page.getByLabel('Approval History')).toBeChecked();
+    await page.getByRole('button', { name: 'Generate Report' }).click();
+});
+
+test("Pop up model opens and updates title on title column value click", async ({ page }) => {
+    await page.goto("https://host.docker.internal/Test_Request_Portal/?a=reports&v=3&query=N4IgLgpgTgtgziAXAbVASwCZJHSAHASQBEQAaEAez2gEMwKpsBCAXjJBjoGMALbKCHAoAbAG4Qs5AOZ0I2AIIA5EgF9S6LIhAYIwiJEmVqUOg2xtynMLyQAGabIXKQKgLrkAVhTQA7BChwwOgBXBHJfNDA0UyhFGhg5dxwGMCRgNRBhNBhIpABGW0LyLJywAHkAMwq4fTsVIA%3D%3D%3D&indicators=NobwRAlgdgJhDGBDALgewE4EkAiYBcYyEyANgKZgA0YUiAthQVWAM4bL4AMAvpeNHCRosuAi2QoAri2a0G%2BMMzboOeHn0iwEKDDgXRiEHeln1Gi6stU8AukA");
+    await page.getByRole('cell', { name: 'Available for test case' }).first().click();
+    await page.getByLabel('Report Title').click();
+
+    await page.getByLabel('Report Title').fill('Available for TEST CHANGE TITLE');
+    await page.getByLabel('Report Title').press('Enter');
+    await page.getByRole('button', { name: 'Save Change' }).click();
+    await page.getByRole('cell', { name: 'Available for TEST CHANGE TITLE' }).first().click();
+    await page.getByLabel('Report Title').click();
+    await page.getByLabel('Report Title').fill('Available for test case');
+    await page.getByLabel('Report Title').press('Enter');
+    await page.getByRole('button', { name: 'Save Change' }).click();
+});
+
+test("Connected forms open on UID link click", async ({ page }) => {
```

Comment:
Maybe better wording: "navigate to a record when clicking on UID link"

The reason "records" is preferred over "forms" in this context is because we commonly need to make a distinction between an empty form and a filled-out record.

Reply:
updated the test name